ledger-core/rms/proto_vnvn_validation_vcachedmemoryresponse.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package rms
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+func (m *VCachedMemoryResponse) Validate(currentPulse PulseNumber) error {
+	object := m.Object.GetValue()
+	_, err := validSelfScopedGlobalWithPulseBeforeOrEq(object, currentPulse, "Object")
+	if err != nil {
+		return err
+	}
+
+	switch m.CallStatus {
+	case CachedMemoryStateUnknown:
+		if m.Object.IsEmpty() {
+			return throw.New("Object should not be empty")
+		}
+		if m.StateID.IsEmpty() {
+			return throw.New("StateID should not be empty")
+		}
+	case CachedMemoryStateFound:
+		if m.Object.IsEmpty() {
+			return throw.New("Object should not be empty")
+		}
+		if m.StateID.IsEmpty() {
+			return throw.New("StateID should not be empty")
+		}
+		if m.Class.IsEmpty() {
```

Comment:
validSelfScopedGlobalWithPulseBeforeOrEq

Reply:
moved to another PR, m.Class removed